net-irc/scrollz/scrollz-2.3.1.ebuild

```diff
@@ -0,0 +1,65 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools toolchain-funcs
+
+MY_P="ScrollZ-${PV}"
+
+DESCRIPTION="Advanced IRC client based on ircII"
+HOMEPAGE="https://www.scrollz.info/"
+SRC_URI="https://www.scrollz.info/download/${MY_P}.tar.gz"
+S="${WORKDIR}/${MY_P}"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~riscv ~x86 ~amd64-linux ~x86-linux ~ppc-macos"
+IUSE="gmp gnutls ipv6 ssl"
+REQUIRED_USE="gnutls? ( ssl )"
+
+BDEPEND="virtual/pkgconfig"
+DEPEND="
+	sys-libs/ncurses:0=
+	virtual/libcrypt:=
+	gmp? ( dev-libs/gmp:0= )
+	ssl? (
+		gnutls? ( net-libs/gnutls:0= )
+		!gnutls? ( dev-libs/openssl:0= )
+	)
+"
+RDEPEND="${DEPEND}"
+
+PATCHES=(
+	"${FILESDIR}/${P}-fix-clang16-build.patch"
```

Comment:
The fix-clang16-build.patch is very large, consider obtaining it via github, e.g. 

```
SRC_URI="
    https://www.scrollz.info/download/${MY_P}.tar.gz
    https://github.com/ScrollZ/ScrollZ/pull/30/commits/344bfdc1e3df6be47971c318dd5492a66e688cd2.patch -> scrollz-2.3.1-fix-clang-and-lto-build.patch
"

PATCHES=(
    # https://github.com/ScrollZ/ScrollZ/pull/30
    "${DESTDIR}/scrollz-2.3.1-fix-clang-and-lto-build.patch"
)

Reply:
Sam usually hosts those when he prams the PR

Reply:
Sure, but if you want to spare him some work, then you could simply use the github link. No strong opinion though, just wanted to point out the possibility.

Reply:
problem is that this patch on the PR on GitHub goes on the HEAD of their master branch, while this here goes on 2.3.1.
There are already a few rejects for that patch from here when I patched their HEAD.
I will correct those on the PR and then this patch won't works on 2.3.1 here anymore.

Reply:
You could create a branch "2.3.1-clang16-and-lto-fixes" containing the patch which applies on 2.3.1 on your github fork on the project and then use this as base for the SRC_URI. I assume that would be the idiomatic thing to do.